homeassistant/components/mqtt/__init__.py

```diff
@@ -1099,6 +1103,8 @@ async def async_added_to_hass(self) -> None:
         """Subscribe MQTT events."""
         await super().async_added_to_hass()
         await self._availability_subscribe_topics()
+        async_dispatcher_connect(self.hass, MQTT_CONNECTED, self.async_mqtt_connect)
+        async_dispatcher_connect(self.hass, MQTT_DISCONNECTED, self.async_mqtt_connect)
```

Comment:
These dispatchers need to be removed when the entity is removed. 

```python
self.async_on_remove(async_dispatcher_connect(self.hass, MQTT_CONNECTED, self.async_mqtt_connect))
```

Reply:
OK, I will create a PR soon.

Reply:
https://github.com/home-assistant/core/pull/37230